### ISSUE TITLE
The unit_count_pattern setting can cause indices to mistakenly be included in an index filter

### ISSUE THREAD

**Issue by soenkeliebau** [elastic/curator#1206]
When defining an age filter with the setting unit_count_pattern and no fallback value the current implementation can cause indices to mistakenly be included or excluded in the filtered list.

Example action:

```
actions:
  1:
    action: delete_indices
    description: Delete indices older than the defined retention time from the index name
    options:
      ignore_empty_list: True
      timeout_override:
      continue_if_exception: False
      disable_action: False
    filters:
    - filtertype: age
      source: name
      direction: older
      timestring: '%Y.%m.%d'
      unit: days
      unit_count: -1
      unit_count_pattern: 'logstash-\w+-([0-9]+)-[0-9]{4}\.[0-9]{2}\.[0-9]{2}'
      exclude:
```

This would work fine for the following index list:
logstash-test-2-2018.04.25
logstash-test-2-2018.04.25
logstash-test-2-2018.04.25
logstash-test-2-2018.04.25

However if you add an index like this:
logstash-te-st-2-2018.04.25
this causes the regular expression to fail once and reverses the decision on all indices that are evaluated afterwards.

The issue comes from [this line of code](https://github.com/elastic/curator/blob/master/curator/indexlist.py#L468) which causes an index that did not match the defined pattern to be ignored, if no fallback value has been configured.
However, since indices are iterated over in a loop exclude stays true for all indices following the non-matching one, causing the exact opposite of the actual intended behavior. This can cause catastrophic data loss if used with a delete option.

I've prepared a fix, will do some additional testing and open a PR shortly.

**Comment by untergeek** [comment]
👍 

For your PR, please ensure that at least one test is added to test this functionality.  I don't care if it's a unit or integration test.

**Comment by untergeek** [comment]
Also, please ensure the PR is against the `5.x` branch.

**Comment by untergeek** [comment]
Fixed by #1208
